contracts/lib/ZoneInteraction.sol

```diff
@@ -118,17 +122,27 @@ contract ZoneInteraction is
             // Set the restricted-order-specific error selector.
             errorSelector = InvalidRestrictedOrder_error_selector;
         } else if (parameters.orderType == OrderType.CONTRACT) {
+            // Set the target to the offerer.
+            target = parameters.offerer;
```

Comment:
`solc` performs a `push20 0xff...ff` masking here.

Reply:
Yeah, this mask is redundant because of the shift later. Would be nice to generalize this and add into the rulelist in solc. Do you want to give it a try @Saw-mon-and-Natalie?

Reply:
But the target value is used below. So maybe it's safe to have the mask (assuming it's there), even though it's used for a `call(...)` later on where the upper bits would be ignored.